### DISCUSSION TITLE
Solid Query and SolidStart

### DISCUSSION THREAD

**Discussion by Blankeos** [orval-labs/orval#1448] [General]
Really cool library! Was wondering if it's possible to port this for **Solid Query** as well? Don't know if I could do it personally but would be very interested in reading through how to do it.

Can anyone point me to the right direction?

**Comment by melloware** [comment]
i would start by reading the source code and taking a look at how Vue Query is generated to make a Solid Query version?

**Comment by AndrejNemec** [comment]
Try this fork of Orval only for Solid.js for now, until the official Orval starts supporting it.

Github: [github.com/AndrejNemec/openapi-tanstack-query-solid#readme](https://github.com/AndrejNemec/openapi-tanstack-query-solid#readme)
NPM: https://www.npmjs.com/package/otqs

**Reply by melloware** [reply]
why not donate a PR?

**Reply by AndrejNemec** [reply]
I did this just for fun and didn’t have a deep understanding of it. I mostly adapted the React Query version to Solid Query. It was just an experiment, but it works well. You can take inspiration from it.

**Reply by melloware** [reply]
I opened this ticket: https://github.com/anymaniax/orval/issues/1486

**Reply by melloware** [reply]
8.0.0-rc6 now supports SolidJS and SolidStart!
